src/main/java/de/tum/cit/aet/artemis/iris/service/pyris/dto/data/PyrisExtendedCourseDTO.java

```diff
@@ -14,25 +14,56 @@
  * An extended course DTO for Pyris so it can better answer
  * questions regarding the course organization and content.
  */
+// @formatter:off
 @JsonInclude(JsonInclude.Include.NON_EMPTY)
-public record PyrisExtendedCourseDTO(long id, String name, String description, Instant startTime, Instant endTime, ProgrammingLanguage defaultProgrammingLanguage,
-        int maxComplaints, int maxTeamComplaints, int maxComplaintTimeDays, int maxRequestMoreFeedbackTimeDays, Integer maxPoints, Integer presentationScore,
-        List<PyrisExerciseWithStudentSubmissionsDTO> exercises, List<PyrisExamDTO> exams, List<PyrisCompetencyDTO> competencies) {
+public record PyrisExtendedCourseDTO(
+        long id,
+        String name,
+        String description,
+        Instant startTime,
+        Instant endTime,
+        ProgrammingLanguage defaultProgrammingLanguage,
+        int maxComplaints,
+        int maxTeamComplaints,
+        int maxComplaintTimeDays,
+        int maxRequestMoreFeedbackTimeDays,
+        Integer maxPoints,
+        Integer presentationScore,
```

Comment:
Since you already added TODOs for api clarity? Can these two attributes be null? Should they then also be wrapped by an optional?

Reply:
I had to check the corresponding Pyris DTO, yes they are optional on Pyris. I think in the PR which handles these DTO inconsistencies we will just have to reevaluate everything anyway, so the existing TODOs are probably good enough.

Reply:
Ah I see, you basically inlined the code :+1: